crates/goose/src/agents/agent.rs

```
@@ -130,7 +126,7 @@ impl Agent {
                 .await
         } else if tool_call.name == PLATFORM_SEARCH_AVAILABLE_EXTENSIONS_TOOL_NAME {
             extension_manager.search_available_extensions().await
-        } else if is_frontend_tool {
+        } else if self.is_frontend_tool(&tool_call.name) {
             // For frontend tools, return an error indicating we need frontend execution
             Err(ToolError::ExecutionError(
                 "Frontend tool execution required".to_string(),
```

why would we want to return an error here? will that cause weird behavior in the agent loop?

---

I think we shouldn't reach this part of the code cause frontend tool requests are handled first, which should be before regular tool requests so my guess is we're trying to be defensive here and bubble up errors in case sth unexpected happened